nearpy/storage/storage_memory.py

```diff
@@ -49,6 +49,14 @@ def store_vector(self, hash_name, bucket_key, v, data):
             self.buckets[hash_name][bucket_key] = []
         self.buckets[hash_name][bucket_key].append((v, data))
 
+    def delete_vector(self, hash_name, data):
+        """
+        Deletes vector and JSON-serializable data in bucket with specified key.
+        """
+        for bucket_key in self.buckets[hash_name]:
+            self.buckets[hash_name][bucket_key] = \
+                [(v, id_data) for v, id_data in self.buckets[hash_name][bucket_key] if id_data != data]
```

Comment:
@erramuzpe I did it by my own.
I personally use [PyCharm](https://www.jetbrains.com/pycharm/). It helps with simple code analysis like pep8 and Python 2/3 compatibility and also allows some simple refactorings. And it has free community version. There are plenty of other linters for Python.
I suggest reading "Writing Idiomatic Python" by Jeff Knupp first, it's an essential book for beginners.
Next you can try [Python Cookbook](http://shop.oreilly.com/product/0636920027072.do). It's more complicated, but it covers almost everything that can be done in Python, including testing.